### ISSUE TITLE
Inconsistent npm run scripts for different environments

### ISSUE THREAD

**Issue by mnigh** [GCTC-NTGC/gc-digital-talent#3768]
## Description
The refresh scripts execute `npm run dev` locally but the deploy script executes `npm run production`. Same goes for the [setup.sh](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/maintenance/scripts/setup.sh#L35) script that executes `npm run dev` and is used in the [e2e-tests github workflow](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/.github/workflows/e2e-tests.yml#L65).
- https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/infrastructure/bin/deploy.sh#L54
- https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/infrastructure/bin/deploy.sh#L61
- https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/infrastructure/bin/deploy.sh#L68
- https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/maintenance/scripts/refresh_admin.sh#L12
- https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/maintenance/scripts/refresh_indigenousapprenticeship.sh#L12
- https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/maintenance/scripts/refresh_talentsearch.sh#L12

### Question
Should the refresh scripts and setup.sh be executing `npm run production` instead of `npm run dev` for consistency across environments?

**Comment by patcon** [comment]
> Should the refresh scripts and setup.sh be executing npm run production instead of npm run dev for consistency across environments?

I understand this different between local and prod environments to be very standard. 

`npm run dev` vs `production` is different because we _do_ want different builds for development -- for example, react component profiling is something one would want in development environment and not in production, and there are some other things that one might want different (sourcemaps, minification). By as practical example, try to open react tab of chrome devtools on production, and the component tree names are all obfuscated, flamegraphs don't work, many helpful console errors are stripped, and I'm sure there are other things

The place where we could catch this would be UAT, which should be built with the same scripts as production.

Having said this, we could def run unit tests and cypress tests on a more prod-like build though, and that's prob doable!

**Comment by mnigh** [comment]
> Having said this, we could def run unit tests and cypress tests on a more prod-like build though, and that's prob doable!

Yes, I think we should. Perhaps we can set an environment variable somewhere in [setup.sh](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/.github/workflows/e2e-tests.yml#L65) that allows to set the `npm run` script as dev or production?

**Comment by patcon** [comment]
Related: This inconsistentcy will now fail tests against `npm run dev` until #4798 is resolved
